api/client/src/main/java/org/projectnessie/client/http/impl/jdk11/JavaHttpClient.java

```diff
@@ -75,6 +76,15 @@ public URI getBaseUri() {
 
   @Override
   public void close() {
+    client
+        .executor()
+        .ifPresent(
+            executor -> {
+              if (executor instanceof ExecutorService) {
+                ((ExecutorService) executor).shutdown();
```

Comment:
Shouldn't we do the traditional dance of `shutdown` + `awaitTermination` + `shutdownNow` ?

Reply:
added.